appcli/configuration_manager.py

```diff
@@ -399,41 +460,71 @@ def __apply_templates_from_directory(
-        """Checks if any template overrides exist in the overrides directory.
+    def __directory_is_not_empty(self, directory: Path) -> bool:
+        """Checks if a directory is not empty.
 
         Returns:
-            bool: Returns True if any overrides exist, otherwise False.
+            bool: Returns True if the directory contains any files, otherwise false
         """
-        template_overrides_dir = self.cli_context.get_template_overrides_dir()
-        return bool(
-            os.path.exists(template_overrides_dir)
-            and os.listdir(template_overrides_dir)
-        )
+        if not os.path.exists(directory):
+            return False
+
+        # Get the files in directory that don't include .gitkeep
+        files_in_directory = os.listdir(directory)
+        try:
+            files_in_directory.remove(".gitkeep")
+        except ValueError:
+            # Ignore if .gitkeep file doesn't exist
+            pass
+
+        return len(files_in_directory) > 0
+
+    def __backup_directory(self, directory_to_backup: Path) -> Path:
+        """Backup a specified directory to a temporary directory
+
+        Args:
+            directory_to_backup (Path): The directory to backup
+
+        Returns:
+            Path: The directory to the temporary backup, or None if the directory to backup doesn't exist
+        """
+        if not os.path.isdir(directory_to_backup):
+            return None
 
-    def __backup_overrides_directory(self) -> Path:
         temp_dir = Path(tempfile.mkdtemp())
 
         # Due to a limitation with 'copytree', it fails to copy if the root directory exists
         # before copying. So we delete the temp dir prior to copying.
         os.rmdir(temp_dir)
-        copy_tree(str(self.cli_context.get_template_overrides_dir()), str(temp_dir))
+        copy_tree(str(directory_to_backup), str(temp_dir))
 
         return temp_dir
 
-    def __copy_in_overrides(self, temp_dir: Path):
-        logger.debug("Copying in overrides")
+    def __copy_directory_contents_with_overwrite(
+        self, source_dir: Path, target_dir: Path
+    ):
+        """Copies the contents of one directory to another, overwriting any existing contents.
+        If the source directory doesn't exist, the target directory will not either.
 
-        overrides_dir = self.cli_context.get_template_overrides_dir()
+        Args:
+            source_dir (Path): Source directory
+            target_dir (Path): Target directory
+        """
+        logger.debug(f"Copying from {source_dir} to {target_dir}")
 
-        copy_tree(str(temp_dir), str(overrides_dir))
-        shutil.rmtree(temp_dir)
+        # If the target directory exists, delete all contents
+        # This is achieved by deleting the folder and re-creating it
+        # This is easier than iterating through the contents and deleting those
+        if os.path.isdir(target_dir):
+            shutil.rmtree(target_dir, ignore_errors=True)
```

Comment:
Right, my earlier comment is wrong then.
We should maybe rename the method. It's more a replace dir function.